implementations/rust/ockam/ockam_node/src/storage/database/migrations/migration_support/migrator.rs

```diff
@@ -228,10 +235,124 @@ impl Migrator {
+                        ),
+                    ));
+                }
+                Ok(false)
+            }
+            None => Ok(true),
+        }
+    }
+
+    async fn apply_sql_migration<'a>(
+        &self,
+        migration: &'a SqlxMigration,
+        connection: &mut AnyConnection,
+        applied_migrations: &[AppliedMigration],
+    ) -> Result<MigrationResult> {
+        if migration.migration_type.is_down_migration() {
+            return Ok(MigrationResult::down_migration());
+        }
+        match applied_migrations
+            .iter()
+            .find(|m| m.version == migration.version)
+        {
+            Some(applied_migration) => {
+                if migration.checksum != applied_migration.checksum {
+                    Ok(MigrationResult::incorrect_checksum(
+                        migration.description.to_string(),
+                        migration.sql.to_string(),
+                        String::from_utf8(migration.checksum.to_vec())
+                            .unwrap_or("actual migration checksum cannot be displayed".to_string()),
+                        String::from_utf8(migration.checksum.to_vec()).unwrap_or(
+                            "expected migration checksum cannot be displayed".to_string(),
+                        ),
+                    ))
+                } else {
+                    Ok(MigrationSuccess)
+                }
+            }
+            None => match connection.apply(migration).await.into_core() {
+                Ok(_) => Ok(MigrationSuccess),
+                Err(e) => Err(ockam_core::Error::new(
+                    Origin::Node,
+                    Kind::Conflict,
+                    format!(
+                        "Failed to run the migration {}: {e:?}",
+                        migration.description
+                    ),
+                )),
+            },
+        }
+    }
+
+    async fn needs_rust_migration<'a>(
+        &self,
+        migration: &'a dyn RustMigration,
+        connection: &mut AnyConnection,
+        _applied_migrations: &[AppliedMigration],
+    ) -> Result<bool> {
+        Ok(!self.has_migrated(connection, migration.name()).await?)
+    }
+
+    async fn apply_rust_migration(
+        &self,
+        migration: &dyn RustMigration,
+        connection: &mut AnyConnection,
+    ) -> Result<MigrationResult> {
+        // If we are migrating data from a legacy sqlite db, we check the SQLite database
+        // to know if that import happened already
+        if migration.name() == InitializeFromSqlite::name() {
+            if let Some(sqlite_db) = &self.legacy_sqlite_database {
+                let mut sqlite_connection = sqlite_db.pool.acquire().await.into_core()?;
+                if !self
+                    .has_migrated(&mut sqlite_connection, migration.name())
+                    .await?
+                {
+                    migration
+                        .migrate(self.legacy_sqlite_database.clone(), connection)
```

Comment:
Just to double-check, the usage of `connection` here instead of `sqlite_connection` is correct here?

Reply:
Yes, we use the `connection` to insert data into Postgres there. 
Then we mark the migration as executed inside the SQLite database.